### PR TITLE
New railtie when Rails is define + More

### PR DESCRIPTION
**Pull request by ericproulx**
Hi,

First of all, Happy new year! I added a couple of thing in your gem.

1. Added a `Railtie` for `Rails`. It seems to be faster to load (well in my app it's way faster based on `bumbler` stats)
2. Updated your travis for multiple ruby versions
3. Added rubocop and run it with auto-correct and auto-gen-config
4. Added `# frozen_literal_string: true`
5. Removed some specs that were not working. It was all about StatementInvalid

Enjoy!

### CODE REVIEW

File: spec/pluck_to_struct_spec.rb [hunk 1]
```diff
@@ -23,15 +23,7 @@
     context 'alias pluck_s' do
       it 'works correctly' do
         struct = TestModel.where(id: 1).pluck_s(:test_attribute).first
-        expect(struct.test_attribute).to eq("test1")
-      end
-    end
-
-    context 'the model does not have the attribute specified' do
```
Comment (girishso) [inline spec/pluck_to_struct_spec.rb]: Please add back this spec

File: spec/support/shared_examples.rb [hunk 2]
```diff
@@ -47,14 +47,6 @@
     end
   end
 
-  context 'the model does not have the attribute specified' do
```
Comment (girishso) [inline spec/support/shared_examples.rb:50]: Please add back this spec

File: spec/support/shared_examples.rb [hunk 3]
```diff
@@ -88,14 +80,6 @@
       end
     end
 
-    context 'the model does not have the attribute specified' do
```
Comment (girishso) [inline spec/support/shared_examples.rb:91]: Please add back this spec

**Review by girishso — CHANGES_REQUESTED** [review]
Hey, Thanks for the PR and Happy New Year!

Good to see addition of rubocop and Railtie.

Everything looks good to me, except the removal of few specs.  Please add back the specs and so I can merge your PR.

**Comment by ericproulx** [comment]
I restored the specs but they ain't passing. Related to Rails 5.1 ?

**Comment by ericproulx** [comment]
Well, the specs ain't working for active record 5.0 and above.

**Comment by ericproulx** [comment]
Hey, have you seen my last message ?

**Pull request closed without merging**